sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala

```diff
@@ -215,6 +237,18 @@ case class OptimizeSkewedJoin(ensureRequirements: EnsureRequirements)
         case (newLeft, newRight) =>
           shj.copy(left = newLeft, right = newRight, isSkewJoin = true)
       }.getOrElse(shj)
+
+    case bhj @ BroadcastHashJoinExec(_, _, _, BuildRight, _,
+        ShuffleStage(left: ShuffleQueryStageExec), _, _, false) =>
```

Comment:
Add `LOCAL_SHUFFLE_READER_ENABLED` check here is better. The BHJ can only be skewed if local shuffle read is disabled. Then we do not need to touch rule `OptimizeShuffleWithLocalRead`.

Please add some comments about why we add this config check.